src/main/java/com/puppycrawl/tools/checkstyle/checks/design/InnerTypeLastCheck.java

```diff
@@ -36,6 +36,48 @@
  * <pre>
  * &lt;module name=&quot;InnerTypeLast&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * class Test { // violation, field should be declared before inner types.
+ *     public void test1() {}
+ *     class InnerTest1 {}
+ *     private String s;
+ *
+ *     class InnerTest2 {
+ *         public void innerTest() {
+ *             int n;
+ *             class LocalClass {
+ *                 public void test3() {}
+ *             }
+ *             int x;
+ *         }
+ *     }
+ *
+ *     public void test() {}
+ *     interface InnerInterface {}
+ *     private int n;
+ * }
+ *
+ * class Test2 { // ok
+ *     private String s;
+ *     private int n;
+ *     public void test() {}
+ *     public void test1() {}
+ *
+ *     class InnerTest1 {
+ *         class InnerTest2 {
```

Comment:
no need for inner inner classes. Keep it simple, just one inner type

Reply:
Done